file.go

```diff
@@ -7,56 +7,109 @@ import (
 )
 
 var (
-	ErrNotDirectory = errors.New("Couldn't call NextFile(), this isn't a directory")
-	ErrNotReader    = errors.New("This file is a directory, can't use Reader functions")
+	ErrNotDirectory = errors.New("file isn't a directory")
+	ErrNotReader    = errors.New("file isn't a regular file")
+
+	ErrNotSupported = errors.New("operation not supported")
 )
 
 // File is an interface that provides functionality for handling
 // files/directories as values that can be supplied to commands. For
-// directories, child files are accessed serially by calling `NextFile()`.
+// directories, child files are accessed serially by calling `NextFile()`
+//
+// Read/Seek methods are only valid for files
+// NextFile method is only valid for directories
 type File interface {
```

Comment:
Purely naming... As a frequent reader of man-pages, I don't mind this naming. However, I've been informed that calling directories files and files regular files is a *bit* confusing. How about `FSObject`, `File`, and `Directory` (or something like that).

Reply:
I agree that the naming is weird, but I'd say that we should keep the base type even if it's just `type Node io.Closer`. The reason is that just using io.Closer would lead to really confusing interfaces - `Add(context.Context, files.Node, ...)` vs `Add(context.Context, io.Closer, ...)` - the second one doesn't relay which interface users should use, will confuse autocompletion in IDEs and won't catch as many errors in compile time.

For names we could use `Node`/`File`/`Directory`

Reply:
Yeah I agree,  I am okay with `Node/File/Directory`.